packages/lib/src/prod/shared-production.ts

```diff
@@ -454,33 +463,58 @@ export function prodSharedPlugin(
         const bIndex = priority.findIndex((value) => value === b[0])
         return aIndex - bIndex
       })
+
+      const manualChunkFunc = (id: string, obj: unknown) => {
+        //  if id is in shared dependencies, return id ,else return vite function value
+        const find = parsedOptions.prodShared.find((arr) =>
+          arr[1].dependencies.has(id)
+        )
+        return find ? find[0] : undefined
+      }
+
       // only active when manualChunks is function,array not to solve
       if (typeof outputOption.manualChunks === 'function') {
         outputOption.manualChunks = new Proxy(outputOption.manualChunks, {
           apply(target, thisArg, argArray) {
-            const id = argArray[0]
-            //  if id is in shared dependencies, return id ,else return vite function value
-            const find = parsedOptions.prodShared.find((arr) =>
-              arr[1].dependencies.has(id)
-            )
-            return find ? find[0] : target(argArray[0], argArray[1])
+            const result = manualChunkFunc(argArray[0], argArray[1])
+            return result ? result : target(argArray[0], argArray[1])
           }
         })
       }
 
+      // The default manualChunk function is no longer available from vite 2.9.0
+      if (typeof outputOption.manualChunks === 'undefined') {
```

Comment:
Why not use `outputOption.manualChunks === undefined` ?

Reply:
done